### PR TITLE
PM_CCCt_366- pr upgrade

### PR DESCRIPTION
**Pull request by pm-dimagi**
https://dimagi.atlassian.net/browse/CCCT-366

## Safety Assurance

- [ ] If the PR is high risk, "High Risk" label is set
- [ ] I have confidence that this PR will not introduce a regression for the reasons below
- [ ] Do we need to enhance manual QA test coverage ? If yes, "QA Note" label is set correctly

### CODE REVIEW

**Review by OrangeAndGreen — CHANGES_REQUESTED** [review]
Getting closer... I'm seeing the phone number picker now in both register and recover scenarios.

There's an issue in the register scenario where the API call is being called too many times (3 instead of 1).

Another issue: in the recovery scenario, when I select a number from the picker, the country code changes from +1 to just 1. This causes a 500 error a bit later when I try to request an OTP from the server using the incorrect number (missing the +).

Also some code cleanup to do.

File: app/src/org/commcare/activities/connect/ConnectIdPhoneActivity.java [hunk 1]
```diff
@@ -90,33 +86,19 @@ void displayNumber(String fullNumber) {
         if (fullNumber != null && fullNumber.startsWith("+" + codeText)) {
             fullNumber = fullNumber.substring(codeText.length() + 1);
         }
-
+        skipPhoneNumberCheck = false;
         uiController.setPhoneNumber(fullNumber);
+        skipPhoneNumberCheck = true;
```
Comment (OrangeAndGreen) [inline app/src/org/commcare/activities/connect/ConnectIdPhoneActivity.java:91]: Should the line below (setCountryCode) be moved up above line 89? Thinking we want to make sure that both the number and country code have been updated to the latest selection before letting the API call trigger

File: app/src/org/commcare/activities/connect/ConnectIdPhoneActivity.java [hunk 2]
```diff
@@ -170,6 +152,7 @@ public void handleButtonPress() {
             IApiCallback callback = new IApiCallback() {
                 @Override
                 public void processSuccess(int responseCode, InputStream responseData) {
+                    skipPhoneNumberCheck=false;
```
Comment (OrangeAndGreen) [inline app/src/org/commcare/activities/connect/ConnectIdPhoneActivity.java:155]: Need to add this for the three error conditions as well

File: app/src/org/commcare/activities/connect/ConnectIdPhoneActivity.java (hunk 1 above)
Comment (pm-dimagi) [reply app/src/org/commcare/activities/connect/ConnectIdPhoneActivity.java:91]: checked this on debug

**Review by OrangeAndGreen — APPROVED** [review]

**Pull request merged**